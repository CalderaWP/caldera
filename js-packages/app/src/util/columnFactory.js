const uuidv1 = require('uuid/v1');

export default function columnFactory(fields = [], columnId=null ) {
	columnId = ! columnId ? uuidv1() : columnId;
	return {
		columnId,
		fields
	}
}
