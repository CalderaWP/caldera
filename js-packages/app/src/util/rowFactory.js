const uuidv1 = require('uuid/v1');
export default function rowFactory(columns = [],rowId=null) {
	rowId = ! rowId ? uuidv1() : rowId;
	return {
		rowId,
		columns
	}
}
