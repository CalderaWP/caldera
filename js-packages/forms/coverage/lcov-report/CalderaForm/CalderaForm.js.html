<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for CalderaForm/CalderaForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">CalderaForm</a> CalderaForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/12</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React from 'react'
import { Formik, Field, Form, ErrorMessage } from 'formik';
import { fieldFactory, fieldGroupFactory } from '@caldera-labs/factory';
import {FormRow} from './components/FormRow';
import {Column,Row,FieldGroup} from '@caldera-labs/factory';
import classNames from "classnames";
&nbsp;
export const CalderaForm = ({ formRows, initialValues,onChange,onBlur }) =&gt; {
	return (
		&lt;div&gt;
			&lt;Formik
				initialValues={initialValues}
				onSubmit={<span class="fstat-no" title="function not covered" >(v</span>alues, actions) =&gt; {
<span class="cstat-no" title="statement not covered" >					setTimeout(<span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >						alert(JSON.stringify(values, null, 2));</span>
<span class="cstat-no" title="statement not covered" >						actions.setSubmitting(false);</span>
					}, 1000);
				}}
				render={({ errors, status, touched, isSubmitting }) =&gt; (
					&lt;Form&gt;
						{formRows.map( (formRow ) =&gt; {
							const {
								rowId,
								columns
							} = formRow;
								return(
									&lt;Row
										className={classNames('caldera-form-row')}
										onChange={onChange}
										onBlur={onBlur}
										key={rowId}
										id={rowId}
									&gt;
										{columns.map( column =&gt; {
											const {
												padding,
												width,
												columnId,
												fields
											} = column;
												return (
													&lt;Column
														key={columnId}
														columnId={columnId}
														width={width}
														padding={padding}
													&gt;
														{fields.map( field  =&gt; {
															const {
																fieldId
															} = field;
															return (
																&lt;FieldGroup
																	key={fieldId}
																	field={field}
																	onChange={onChange}
																	onBlur={onBlur}
																	fieldErrors={errors}
																	fieldsTouch={touched}
																/&gt;
															)
														})}
													&lt;/Column&gt;
												)
										})}
									&lt;/Row&gt;
								)
&nbsp;
&nbsp;
						})}
						&lt;button type="submit" disabled={isSubmitting}&gt;
							Click Button
						&lt;/button&gt;
					&lt;/Form&gt;
				)}
			/&gt;
		&lt;/div&gt;
	);
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 31 2018 20:36:55 GMT-0500 (EST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
