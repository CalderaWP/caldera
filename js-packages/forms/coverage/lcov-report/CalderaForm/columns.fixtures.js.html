<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for CalderaForm/columns.fixtures.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">CalderaForm</a> columns.fixtures.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField,
	radioField
} from './fields.fixtures';
import React from 'react';
&nbsp;
export const formRowOne = {
	rowId: 'r1',
	columns: [
		{
			fields: [emailField],
			width: '1/2',
			columnId: '1a'
		},
		{
			fields: [textField],
			width: '1/4',
			columnId: '1b'
		}
	]
};
&nbsp;
export const formRowTwo = {
	rowId: 'r2',
	columns: [
		{
			fields: [checkboxField, selectField],
			width: '1/2',
			columnId: 1
		},
		{
			fields: [radioField],
			width: '1/4',
			columnId: 2
		}
	]
};
&nbsp;
export const formRows = [formRowOne, formRowTwo];
&nbsp;
export const notFormRow = {
	rowId: 'notForm',
	columns: [
		{
			children: &lt;div&gt;Half Column&lt;/div&gt;,
			width: '1/2',
			columnId: 1
		},
		{
			children: &lt;div&gt;Quarter Column&lt;/div&gt;,
			width: '1/4',
			columnId: 2
		},
		{
			children: &lt;div&gt;Quarter Column 2&lt;/div&gt;,
			width: '1/4',
			columnId: 3
		}
	]
};
&nbsp;
export const notFormRows = [notFormRow];
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 02 2019 19:23:31 GMT-0500 (EST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
