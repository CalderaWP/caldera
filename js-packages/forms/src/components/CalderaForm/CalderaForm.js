import PropTypes from 'prop-types';
import React from 'react';
import { Formik, Field, Form, ErrorMessage } from 'formik';
import { fieldFactory, fieldGroupFactory } from '@caldera-labs/factory';
import { Column, Row, FieldGroup } from '@caldera-labs/factory';
import classNames from 'classnames';

export const CalderaForm = ({ formRows, initialValues, onSubmit }) => {
	return (
		<div>
			<Formik
				initialValues={initialValues}
				onSubmit={onSubmit}
				render={({
					errors,
					status,
					touched,
					isSubmitting,
					handleChange,
					handleBlur,
					setFieldValue,
					values
				}) => (
					<Form>
						{formRows.map(formRow => {
							const { rowId, columns } = formRow;
							return (
								<Row
									className={classNames('caldera-form-row')}
									key={rowId}
									id={rowId}
								>
									{columns.map(column => {
										const {
											padding,
											width,
											columnId,
											fields
										} = column;
										return (
											<Column
												key={columnId}
												columnId={columnId}
												width={width}
												padding={padding}
											>
												{fields.map(field => {
													const { fieldId } = field;
													field.value =
														values[fieldId];
													return (
														<FieldGroup
															key={fieldId}
															field={field}
															onChange={newValue => {
																setFieldValue(
																	fieldId,
																	newValue,
																	true
																);
															}}
															onBlur={handleBlur}
															fieldErrors={errors}
															fieldsTouch={
																touched
															}
														/>
													);
												})}
											</Column>
										);
									})}
								</Row>
							);
						})}
						<input
							type="submit"
							className={'caldera-forms-submit'}
							disabled={isSubmitting}
							value={'Click Button'}
						/>
					</Form>
				)}
			/>
		</div>
	);
};

CalderaForm.propTypes = {
	formRows: PropTypes.array,
	initialValues: PropTypes.object,
	onSubmit: PropTypes.func
};
