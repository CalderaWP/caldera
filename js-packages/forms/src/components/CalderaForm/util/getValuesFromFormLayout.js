import {collectFieldValues} from "@caldera-labs/factory";

export const getValuesFromFormLayout = (rows) => {
	let values=  {

	};
	rows.map( formRow => {
		const {
			columns
		} = formRow;
		columns.map( column => {
			const {
				fields
			} = column;
			values = {
				...values,
				...collectFieldValues(fields)
			}
		});

	});
	return values;
};
