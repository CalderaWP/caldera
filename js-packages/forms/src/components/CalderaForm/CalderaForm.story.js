import React from 'react';
import { storiesOf } from '@storybook/react';
import { CalderaForm } from './CalderaForm';
import { formRows } from './columns.fixtures';

import { collectFieldValues } from '@caldera-labs/factory';

import { getValuesFromFormLayout } from './util/getValuesFromFormLayout';

const initialValues = getValuesFromFormLayout(formRows);
storiesOf('CalderaForm', module).add('Forms', () => (
	<CalderaForm
		formRows={formRows}
		initialValues={initialValues}
		onSubmit={(values, actions) => {
			setTimeout(() => {
				alert(JSON.stringify(values, null, 2));
				actions.setSubmitting(false);
			}, 1000);
		}}
	/>
));
