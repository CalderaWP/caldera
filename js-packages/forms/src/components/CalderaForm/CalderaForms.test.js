import { CalderaForm } from './CalderaForm';
import React, {Fragment} from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { formRows,formRowOne,checkboxField } from './columns.fixtures';
import { getValuesFromFormLayout } from './util/getValuesFromFormLayout';
import {emailField, textField} from "./fields.fixtures";

const initialValues = getValuesFromFormLayout(formRows);
describe('Caldera Forms', () => {
	let onChange = jest.fn();
	let onBlur = jest.fn();
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('Forms', () => {
		const component = renderer.create(
			<CalderaForm
				formRows={formRows}
				initialValues={initialValues}
				onBlur={onBlur}
				onChange={onChange}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'A row can have a render prop', () => {
		const _Row = (props) => <div id={'test2243'}/>;

		const component = mount(
			<CalderaForm
				formRows={[
					formRowOne,
					{
						rowId: 'r45',
						render:  _Row
					}
				]}
				initialValues={initialValues}
				onBlur={onBlur}
				onChange={onChange}
			/>
		);
		expect(component.find('#test2243').length).toBe(1);
	});
	it( 'A column can have a render prop', () => {
		const _Field = (props) => <input id={'test243'} type={'number'} key={808}/>;

		const component = mount(
			<CalderaForm
				formRows={[
					{
						rowId: 'r1',
						columns: [
							{
								render: _Field,
								width: '1/2',
								columnId: '1aaaaa'
							},
							{
								fields: [textField],
								width: '1/4',
								columnId: '1b'
							}
						]
					}
				]}
				initialValues={initialValues}
				onBlur={onBlur}
				onChange={onChange}
			/>
		);
		expect(component.find('#test243').length).toBe(1);
	});

	it( 'A field can have a render prop', () => {
		const _Field = (props) => <input id={'test808'} type={'number'} />;

		const field = {
			...checkboxField,
			render: _Field,
			key:800000
		};
		const component = mount(
			<CalderaForm
				formRows={[
					{
						rowId: 'r1',
						columns: [
							{
								fields: [
									field,
									checkboxField
								],
								width: '1/2',
								columnId: '1aaaaa'
							},
							{
								fields: [textField],
								width: '1/4',
								columnId: '1b'
							}
						]
					}
				]}
				initialValues={initialValues}
				onBlur={onBlur}
				onChange={onChange}
			/>
		);
		expect(component.find('#test808').length).toBe(1);
		expect(component.find('#test808').prop('type')).toBe('number');
	});


});


describe('Updates field value', () => {
	let onSubmit = jest.fn();
	beforeEach(() => {
		onSubmit = jest.fn();
	});

	it('Updates field value when field is changed', () => {
		const component = mount(
			<CalderaForm
				formRows={formRows}
				initialValues={initialValues}
				onSubmit={onSubmit}
			/>
		);
		const value = 'changed@email22.com';
		const event = { target: { value } };
		component
			.find('#emailFieldId')
			.find('input')
			.simulate('change', event);
		expect(
			component
				.find('#emailFieldId')
				.find('input')
				.prop('value')
		).toEqual(value);
	});

	it( 'Recives values update when any field changes', () => {
		const _onChange = jest.fn();
		const component = mount(
			<CalderaForm
				formRows={formRows}
				initialValues={initialValues}
				onSubmit={onSubmit}
				onChange={_onChange}
			/>
		);
		const value = 'changed@email22.com';
		const event = { target: { value } };
		component
			.find('#emailFieldId')
			.find('input')
			.simulate('change', event);
		expect(
			_onChange.mock.calls.length
		).toEqual(1);

	})
});

