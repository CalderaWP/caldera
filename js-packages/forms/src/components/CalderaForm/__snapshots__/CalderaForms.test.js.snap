// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caldera Forms Forms 1`] = `
<div>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="sc-bdVaJa caldera-row caldera-form-row sc-bwzfXH pjOvZ"
      id={undefined}
    >
      <div
        className="caldera-column sc-bdVaJa gfmZBt"
        id="1a"
        width="1/2"
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control caldera-field caldera-field-email"
          >
            <div
              className="components-base-control__field"
            >
              <label
                className="components-base-control__label"
                htmlFor="inspector-text-control-0"
              >
                Email Field Label
              </label>
              <input
                aria-describedby="inspector-text-control-0__help"
                className="components-text-control__input"
                id="emailFieldId"
                multiple={true}
                onBlur={[Function]}
                onChange={[Function]}
                placeholder={undefined}
                type="email"
                value="emai@email.com"
              />
            </div>
            <p
              className="components-base-control__help"
              id="inspector-text-control-0__help"
            >
              Email field description
            </p>
          </div>
        </div>
      </div>
      <div
        className="caldera-column sc-bdVaJa jlBFAP"
        id="1b"
        width="1/4"
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control caldera-field caldera-field-text"
          >
            <div
              className="components-base-control__field"
            >
              <label
                className="components-base-control__label"
                htmlFor="inspector-text-control-1"
              >
                First Name
              </label>
              <input
                aria-describedby="inspector-text-control-1__help"
                className="components-text-control__input"
                id="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder={undefined}
                type="text"
                value="roy"
              />
            </div>
            <p
              className="components-base-control__help"
              id="inspector-text-control-1__help"
            >
              your first name
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa caldera-row caldera-form-row sc-bwzfXH pjOvZ"
      id={undefined}
    >
      <div
        className="caldera-column sc-bdVaJa gfmZBt"
        id={1}
        width="1/2"
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control"
          >
            <div
              className="components-base-control__field"
            >
              <input
                aria-describedby="inspector-checkbox-control-0__help"
                checked={true}
                className="components-checkbox-control__input"
                id="checkboxFieldLabel"
                onBlur={[Function]}
                onChange={[Function]}
                type="checkbox"
                value="1"
              />
              <label
                className="components-checkbox-control__label"
                htmlFor="inspector-checkbox-control-0"
              >
                Checkbox Labe
              </label>
            </div>
            <p
              className="components-base-control__help"
              id="inspector-checkbox-control-0__help"
            >
              Checkbox field description
            </p>
          </div>
        </div>
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control caldera-field caldera-field-select"
          >
            <div
              className="components-base-control__field"
            >
              <label
                className="components-base-control__label"
                htmlFor="inspector-select-control-0"
              >
                Select Field Label
              </label>
              <select
                aria-describedby="inspector-select-control-0__help"
                className="components-select-control__input"
                id="selectFieldId"
                multiple={false}
                onChange={[Function]}
                value={1}
              >
                <option
                  value={0}
                >
                  Zero
                </option>
                <option
                  value={1}
                >
                  One
                </option>
                <option
                  value={2}
                >
                  Two
                </option>
              </select>
            </div>
            <p
              className="components-base-control__help"
              id="inspector-select-control-0__help"
            >
              Select field description
            </p>
          </div>
        </div>
      </div>
      <div
        className="caldera-column sc-bdVaJa jlBFAP"
        id={2}
        width="1/4"
      >
        <div
          className="caldera-field-wrapper caldera-field-wrapper-text"
        >
          <div
            className="components-base-control caldera-field caldera-field-radio components-radio-control"
          >
            <div
              className="components-base-control__field"
            >
              <label
                className="components-base-control__label"
                htmlFor="inspector-radio-control-0"
              >
                Radio Field Label
              </label>
              <div
                className="components-radio-control__option"
              >
                <input
                  aria-describedby="inspector-radio-control-0__help"
                  checked={false}
                  className="components-radio-control__input"
                  id="inspector-radio-control-0-0"
                  name="inspector-radio-control-0"
                  onChange={[Function]}
                  type="radio"
                  value={0}
                />
                <label
                  htmlFor="inspector-radio-control-0-0"
                >
                  Zero
                </label>
              </div>
              <div
                className="components-radio-control__option"
              >
                <input
                  aria-describedby="inspector-radio-control-0__help"
                  checked={true}
                  className="components-radio-control__input"
                  id="inspector-radio-control-0-1"
                  name="inspector-radio-control-0"
                  onChange={[Function]}
                  type="radio"
                  value={1}
                />
                <label
                  htmlFor="inspector-radio-control-0-1"
                >
                  One
                </label>
              </div>
              <div
                className="components-radio-control__option"
              >
                <input
                  aria-describedby="inspector-radio-control-0__help"
                  checked={false}
                  className="components-radio-control__input"
                  id="inspector-radio-control-0-2"
                  name="inspector-radio-control-0"
                  onChange={[Function]}
                  type="radio"
                  value={2}
                />
                <label
                  htmlFor="inspector-radio-control-0-2"
                >
                  Two
                </label>
              </div>
            </div>
            <p
              className="components-base-control__help"
              id="inspector-radio-control-0__help"
            >
              Radio field description
            </p>
          </div>
        </div>
      </div>
    </div>
    <input
      className="caldera-forms-submit"
      disabled={false}
      type="submit"
      value="Click Button"
    />
  </form>
</div>
`;
